src/main/java/net/minecraftforge/server/permission/PermissionAPI.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Minecraft Forge
+ * Copyright (c) 2016.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation version 2.1
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package net.minecraftforge.server.permission;
+
+import com.google.common.base.Preconditions;
+import com.mojang.authlib.GameProfile;
+import net.minecraftforge.fml.common.FMLLog;
+import net.minecraftforge.server.permission.context.IContext;
+import org.apache.logging.log4j.Level;
+
+import javax.annotation.Nonnull;
+
+public class PermissionAPI
+{
+    private static IPermissionHandler permissionHandler = DefaultPermissionHandler.INSTANCE;
+
+    public static void setPermissionHandler(@Nonnull IPermissionHandler handler)
```

Comment:
There's nothing right either. What if I have, let's say, mod A, that provide permissions, but I want to write a mod B that enhances certain aspect of mod A? I.e. provide a more fine grained or more user-friendly area control?
With normal implementation, I would store handler instaled by mod A, register my mod B, and pass through any checks except those I'm interested in. Ultimately this means I could make mod A optional, but my mod would then only provide what it can, and the rest will be deferred to whatever permission handler was there before registration.
You'd find examples of such implementation everywhere, even in OS itself.